warehouse/query-core/src/main/java/datawave/query/tables/ShardQueryLogic.java

```diff
@@ -1344,7 +1395,163 @@ public void close() {
                 log.error("Caught exception trying to close Scheduler", e);
             }
         }
+    }
+
+    @Override
+    public Object validateQuery(AccumuloClient client, Query settings, Set<Authorizations> auths, boolean expandFields, boolean expandValues) throws Exception {
+        this.config = ShardQueryConfiguration.create(this, settings);
+        if (log.isTraceEnabled()) {
+            log.trace("Initializing ShardQueryLogic for query validation: " + System.identityHashCode(this) + '('
+                            + (this.getSettings() == null ? "empty" : this.getSettings().getId()) + ')');
+        }
+
+        // todo - maybe unnecessary, we could just return early if no rules configured.
+        if (validationRules == null || validationRules.isEmpty()) {
+            throw new IllegalStateException("Query validation rules not configured.");
+        }
+
+        // todo - verify if we should allow these to be configured, or always stick to an established default.
+        this.config.setExpandFields(expandFields);
```

Comment:
I am confused here.  What does expanding fields and values have to do with the query validation?  I do not believe those concepts are used at all.  They only apply to query planning which we are not doing here.  We are only validating the initial query.  Am I missing something?